services/user/Subgroups/service/src/lib.rs

```diff
@@ -0,0 +1,104 @@
+#[psibase::service(name = "subgroups")]
+pub mod service {
+    pub use psibase::fracpack;
+
+    /// Function that determines which subgrouping solution is preferred based on specific criteria.
+    /// Returns true if the new solution is preferred over the old solution.
+    type PreferenceFunction = fn(new: &[u32], old: &[u32]) -> bool;
+
+    /// Prefers groupings with the greatest minimum group size.
+    /// Achieves by comparison based on lexicographical ordering.
+    /// Returns true if new > old.
+    fn greatest_minimum_preference(new: &[u32], old: &[u32]) -> bool {
+        let mut new_sorted = new.to_vec();
+        let mut old_sorted = old.to_vec();
+
+        new_sorted.sort();
+        old_sorted.sort();
+
+        for (n, o) in new_sorted.iter().zip(old_sorted.iter()) {
```

Comment:
https://doc.rust-lang.org/std/vec/struct.Vec.html#impl-Ord-for-Vec%3CT,+A%3E

Reply:
Oh, nice. Simplified